exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestNestedLoopJoin.java

```diff
@@ -333,4 +341,74 @@ public void testNlJoinWithLargeRightInputSuccess() throws Exception {
       test(RESET_JOIN_OPTIMIZATION);
     }
   }
+
+  private void buildFile(String fileName, String[] data, File testDir) throws IOException {
+    try(PrintWriter out = new PrintWriter(new FileWriter(new File(testDir, fileName)))) {
+      for (String line : data) {
+        out.println(line);
+      }
+    }
+  }
+
+  @Test
+  public void testNestedLeftJoinWithEmptyTable() throws Exception {
+    setSessionOption("planner.enable_join_optimization", false);
+    setSessionOption("planner.enable_nljoin_for_scalar_only", false);
+    setSessionOption("planner.enable_hashjoin", false);
+    setSessionOption("planner.enable_mergejoin", false);
+    try (ClusterFixture cluster = ClusterFixture.builder(dirTestWatcher).build();
+      ClientFixture client = cluster.clientFixture()) {
+      File testDir = dirTestWatcher.getRootDir();
+      buildFile("dept.json", new String[0], testDir);
+      QueryBuilder query = client.queryBuilder().sql("select * from cp.`employee.json` emp left outer join dfs.`dept.json` as dept on dept.manager = emp.`last_name`");
+      assert(query.run().recordCount() == 1155);
+    } catch (RuntimeException ex) {
+      fail (ex.getMessage());
+    } finally {
+      resetSessionOption("planner.enable_join_optimization");
+      resetSessionOption("planner.enable_nljoin_for_scalar_only");
+      resetSessionOption("planner.enable_hashjoin");
+      resetSessionOption("planner.enable_mergejoin");
+    }
+  }
+
+  @Test
+  public void testNestedInnerJoinWithEmptyTable() throws Exception {
+    setSessionOption("planner.enable_nljoin_for_scalar_only", false);
```

Comment:
Same comment as above: duplicate code. Here we see the problem with copies. This test, and the next one, set three options. The prior one sets four options. Which is right?